populator-machinery/controller.go

```diff
@@ -189,6 +194,21 @@ type PopulatorParams struct {
 	Recorder     record.EventRecorder
 }
 
+type MutatorConfig struct {
+	// PvcPrimeMutator is the mutator function for pvcPrime
+	PvcPrimeMutator func(MutatorParams) error
+}
+
+// MutatorParams includes the parameters passing to mutator functions
```

Comment:
Are all these intended to be mutated? I think just the PvcPrime is intended to be mutated, right?

If so, please return it as an output variable, and note that these variable Pvc and StorageClass are inputs, so it's clear to the user that they aren't expected to be mutated.

Reply:
My thinking of using MutatorParams instead of PVC and StorageClass as input, is to leave some room for future extend, if others need more information, they can just add new fields in the struct?

Reply:
For the return value, if we pass a pointer for pvcPrime as input, what would be the best practice of a modification function? I feel input pvcPrime, modify, return and assign to the same pvcPrime seems not necessary?

Reply:
Discussed offline, we will use `PvcPrimeMutatorParams` as the input to avoid implicit contract problem and make it extendable with future changes, and use (*corev1.PersistentVolumeClaim, error) as output follow the best practice of mutator functions in go, an example in client-go  https://pkg.go.dev/k8s.io/client-go@v0.32.0/tools/cache#TransformFunc